frontend/app/src/components/StreamlitDialog/StreamlitDialog.test.tsx

```diff
@@ -38,13 +40,16 @@ describe("StreamlitDialog", () => {
     // Flush promises to give componentDidMount() a chance to run.
     await flushPromises()
 
-    setTimeout(() => {
-      expect(wrapper.find("button").at(1).is(":focus")).toBe(true)
-    }, 0)
+    await waitFor(() => {
+      const buttons = screen.getAllByRole("button")
+      const targetButton = buttons[1]
+
+      expect(targetButton).toHaveTextContent("Clear caches")
+    })
```

Comment:
Can we add the focus check as well? Can probably use [`.toHaveFocus`](https://www.npmjs.com/package/@testing-library/jest-dom#tohaveFocus)

Reply:
done